tableau_utilities/hyper/hyper.py

```diff
@@ -0,0 +1,112 @@
+import os
+import shutil
+from zipfile import ZipFile
+from tableauhyperapi import HyperProcess, Connection, Telemetry, CreateMode, TableDefinition, TableName, SqlType
+
+from tableau_utilities.tableau_file.tableau_file import TableauFileError, Datasource
+
+
+def create_empty_hyper_extract(datasource: Datasource):
+    """ Creates an empty extract (.hyper file) for the Tableau file.
+        If the extract exists, it will be overwritten.
+        Args:
+            Datasource: The tableau_utilities datasource class
+    """
+    # Get relevant paths, and create a temp folder and move the Tableau file into it
+    temp_folder = os.path.join(datasource.file_directory, f'__TEMP_{Datasource.file_name}')
+    extract_folder = os.path.join(temp_folder, 'Data', 'Extracts')
+    hyper_rel_path = os.path.join('Data', 'Extracts', f'{Datasource.file_name}.hyper')
+    temp_path = os.path.join(temp_folder, Datasource.file_basename)
+    tdsx_basename = f'{Datasource.file_name}.tdsx'
+    tdsx_path = os.path.join(temp_folder, tdsx_basename)
+    os.makedirs(extract_folder, exist_ok=True)
+    shutil.move(Datasource.file_path, temp_path)
+    if Datasource.extension == 'tdsx':
+        # Unzip the TDS file
+        with ZipFile(temp_path) as z:
+            for f in z.filelist:
+                ext = f.filename.split('.')[-1]
+                if ext in ['tds', 'twb']:
+                    tds_path = z.extract(member=f, path=temp_folder)
+    else:
+        tds_path = temp_path
+    hyper_path = os.path.join(extract_folder, f'{Datasource.file_name}.hyper')
+    params = {"default_database_version": "2"}
+    # Get columns from the metadata
+    columns = dict()  # Use a dict to ensure no duplicate columns are referenced
+    for metadata in Datasource.connection.metadata_records:
```

Comment:
```suggestion
    temp_path = os.path.join(temp_folder, datasource.file_basename)
    tdsx_basename = f'{datasource.file_name}.tdsx'
    tdsx_path = os.path.join(temp_folder, tdsx_basename)
    os.makedirs(extract_folder, exist_ok=True)
    shutil.move(datasource.file_path, temp_path)
    if datasource.extension == 'tdsx':
        # Unzip the TDS file
        with ZipFile(temp_path) as z:
            for f in z.filelist:
                ext = f.filename.split('.')[-1]
                if ext in ['tds', 'twb']:
                    tds_path = z.extract(member=f, path=temp_folder)
    else:
        tds_path = temp_path
    hyper_path = os.path.join(extract_folder, f'{datasource.file_name}.hyper')
    params = {"default_database_version": "2"}
    # Get columns from the metadata
    columns = dict()  # Use a dict to ensure no duplicate columns are referenced
    for metadata in datasource.connection.metadata_records:
```
Sorry @jaybythebay I mean you'll have to go in and update ever one of these `Datasource` references in both of these functions, to use the lowercase alias you renamed for the arg.

I didn't include them all in this suggestion, so you'll have to manually update the rest yourself.